.github/workflows/yaml-pr.yml

```diff
@@ -0,0 +1,116 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Checks that are intended to run on PRs containing Java code.
+
+name: Java PR
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+    paths:
+      - '**.java'
+      - '**.xml'
+      # Include python files and Dockerfiles used for YAML and xlang templates.
+      - '**.py'
+      - 'plugins/core-plugin/src/main/resources/**'
+      # Include relevant GitHub Action files for running these checks.
+      # This will make it easier to verify action changes don't break anything.
+      - '.github/actions/setup-env/*'
+      - '.github/workflows/java-pr.yml'
```

Comment:
Do we actually need all of these paths?

Reply:
It seems like this should be scoped to yaml

Reply:
Also, is this workflow a subset of the java-pr workflow? If so, we should get rid of the checkstyle/spotless checks below